### ISSUE TITLE
-rifrom not failing with Coq 8.18.0

### ISSUE THREAD

**Issue by SnarkBoojum** [coq/coq#18592]
I was trying to write some code to ease work on packaging the Coq ecosystem for Debian, but was surprised to have a hard time detecting problems, until I realized something was amiss:
```
jpuydt@alibert:~$ coqc -rifrom foo bar && echo "Ok"
Ok
jpuydt@alibert:~$ coqtop
Welcome to Coq 8.18.0

Coq < From foo Require Import bar.
Toplevel input, characters 0-28:
> From foo Require Import bar.
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: Cannot find a physical path bound to logical path bar with prefix foo.

Coq < 
```

I would have expected the batch compilation to fail with a stderr output instead of silently returning 0.

**Comment by SkySkimmer** [comment]
When not given a file to compile coqc won't try to run the requires. For instance `coqc -rifrom nowhere notreal empty.v` does fail.

Maybe we should have an error when coqc is called with no main argument?

**Comment by SnarkBoojum** [comment]
Yes, or take the command-line arguments into account (possibly giving an error) before turning to the list of files to compile, and not consider an empty list a problem...

That made my code a bit more complex since I have to create a temporary directory and an empty file, see [here](https://salsa.debian.org/ocaml-team/dh-coq/-/blob/master/tools/coq-check-packages).

**Comment by SkySkimmer** [comment]
>Yes, or take the command-line arguments into account (possibly giving an error) before turning to the list of files to compile, and not consider an empty list a problem...

Not really possible for -rifrom currently, we need to be processing a file before we can do Require

**Comment by SkySkimmer** [comment]
>That made my code a bit more complex since I have to create a temporary directory and an empty file, see here.

(You could use coqtop instead of coqc for this, eg `coqtop -rifrom foo bar < /dev/null`)
